### ISSUE TITLE
Nomad's native service registration fails if raw_exec and docker drivers are combined in the same task group

### ISSUE THREAD

**Issue by groggemans** [hashicorp/nomad#13483]
### Nomad version
Nomad v1.3.1 (2b054e38e91af964d1235faa98c286ca3f527e56)

### Operating system and Environment details
Linux + docker

### Issue
Using native service registration fails if a raw_exec and docker tasks are combined.
(using two docker tasks in the same task group isn't an issue)

### Reproduction steps
Below example jobs runs two counting loops in a sh script, one in a container, the other directly on the host.
There is a dummy port and service registration that's not actually used, but that does not make a difference. (The initial discovery was with a job that actually used the port, but it's not needed to replicate the behavior)
I discovered it with a pretask hook as the raw_exec task, but using an actually long living task seems to have the same effect.

#### Expected Result
The service is registered and can be checked with `nomad service list`.

#### Actual Result
The service fails to register with below error;

```
Task hook failed | task_services: rpc error: service registration insert failed: object missing primary index
```

### Job file (if appropriate)

```hcl
job "example" {
  datacenters = ["dc1"]

  group "counter" {
    network {
      port "random" {
      }
    }

    task "raw" {
      driver = "raw_exec"
      config {
            command = "${NOMAD_TASK_DIR}/my-script.sh"
      }
      template {
        destination = "${NOMAD_TASK_DIR}/my-script.sh"
        perms = "755"
        data = <<EOF
#!/bin/sh
i=0

while true; do
  echo "$i"
  i=$((i+1))
  sleep 2
done

EOF
      }
    }

    task "docker" {
      driver = "docker"
      config {
        image = "alpine:3.13.4"
        command = "${NOMAD_TASK_DIR}/my-script.sh"
        ports = ["random"]
      }
      service {
        provider = "nomad"
        name = "counter"
        port = "random"
      }

      template {
        destination = "${NOMAD_TASK_DIR}/my-script.sh"
        perms = "755"
        data = <<EOF
#!/bin/sh
i=0

while true; do
  echo "$i"
  i=$((i+1))
  sleep 2
done

EOF
      }
    }
  }
}
```

**Comment by attachmentgenie** [comment]
@jrasell is this supposed to be part of 1.3.2? in any case i am still observing it there too

**Comment by lgfa29** [comment]
> is this supposed to be part of 1.3.2?

Ah sorry for that. Yes, the initial plan was to have it shipped in 1.3.2 but the backport never happened, so this was shipped in 1.3.4. 

Apologies for the confusion.

**Comment by attachmentgenie** [comment]
i can confirm this now works in 1.3.5
